packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -48,25 +49,92 @@ Future<void> updateGeneratedXcodeProperties({
   bool useMacOSConfig = false,
   bool setSymroot = true,
 }) async {
+  final List<String> xcodeBuildSettings = _xcodeBuildSettingsLines(
+    project: project,
+    buildInfo: buildInfo,
+    targetOverride: targetOverride,
+    useMacOSConfig: useMacOSConfig,
+    setSymroot: setSymroot
+  );
+
+  _updateGeneratedXcodePropertiesFile(
+    project: project,
+    xcodeBuildSettings: xcodeBuildSettings,
+    useMacOSConfig: useMacOSConfig,
+  );
+
+  _updateGeneratedModuleBuildPhaseScript(
```

Comment:
New script looks like
```
#!/bin/sh
# This is a generated file; do not edit or check into version control.
export FLUTTER_ROOT=/path/to/flutter
export FLUTTER_APPLICATION_PATH=/path/to/host/my_flutter
export FLUTTER_TARGET=lib/main.dart
export FLUTTER_BUILD_DIR=build
export SYMROOT=${SOURCE_ROOT}/../build/ios
export FLUTTER_BUILD_NAME=1.0.0
export FLUTTER_BUILD_NUMBER=1
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
```

Reply:
No, it gets created when the module gets created or re-created when the module is made editable.  It's available immediately.
Do you know what processes to people tack on that use these variables?  I'm curious why they would need FLUTTER_ variables.

Reply:
> Do you know what processes to people tack on that use these variables? I'm curious why they would need FLUTTER_ variables. 

I'm not sure I understood the question :S

Reply:
> If so, I think it's right but it's a breaking change (in case people tacked on other processes that depends on these variables). 

Do you have an example?  Or is that speculative?

Reply:
It's pure speculation. I don't mean that we need to accommodate them or do anything special. Just announce it so people whose toolchain breaks don't have to dig around solving a mystery for an hour.

Reply:
I think the remaining comment here is can this be in a template file

Reply:
I believe the templates are never edited after their first creation/making the module editable unless the file is missing.  The only variables that are filled out in the template are things the tool knows at the time of creation, like the projectName.
The generated files like this new one and the old Generated.xcconfig are re-generated on every build.  FLUTTER_FRAMEWORK_DIR gets updated to the right engine build based on the build configuration (debug, profile, release), and the build name and number are re-read from the manifest, etc.

Reply:
And this comment made me realize that we can probably parse out the FLUTTER_APPLICATION_PATH in the podhelper just like it parsed FLUTTER_ROOT to find the xcode_backend script, so that path doesn't need to get passed in from the host Podfile.

Reply:
Oh never mind, it's needed to find the podhelper script in the first place.

Reply:
Ah interesting. I never thought about the distinction. Could be worth adding a comment here.